doc/reference/reference_sql/sql_user_guide.rst

```diff
@@ -555,17 +613,45 @@ Prior to Tarantool version 2.10.1, individual column values had
 one of the preceding types -- BOOLEAN, INTEGER, DOUBLE, DECIMAL, STRING, VARBINARY, or UUID.
 Starting in Tarantool version 2.10.1, all values have type SCALAR.
 
+.. _sql_data_type_map:
+
+MAP values are key:value combinations which can be produced with
+:ref:`MAP expressions <sql_map_expression>`.
+Maps cannot be used in arithmetic or comparison (except ``IS [NOT] NULL``),
+and the only
+functions where they are allowed are :ref:`CAST <sql_function_cast>`,
+:ref:`QUOTE <sql_function_quote>`,
+:ref:`TYPEOF <sql_function_typeof>`, and functions involving NULL comparisons.
+
+.. _sql_data_type_array:
+
+ARRAY values are lists which can be produced with
+:ref:`ARRAY expressions <sql_array_expression>`.
+Arrays cannot be used in arithmetic or comparison (except ``IS [NOT] NULL``), and the only
+functions where they are allowed are :ref:`CAST <sql_function_cast>`,
+:ref:`QUOTE <sql_function_quote>`,
+:ref:`TYPEOF <sql_function_typeof>`, and functions involving NULL comparisons.
+
+.. _sql_data_type_any:
+
+ANY can be used for
+:ref:`column definitions <sql_column_def_data_type>` and the individual column values have
+type ANY.
+The difference between SCALAR and ANY is:
+SCALAR columns may not contain MAP or ARRAY values, but ANY columns may contain them.
```

Comment:
I think we should add that ANY values are not comparable (SCALAR values are comparable).